frontends/web/src/hooks/localized.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2024 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useMemo } from 'react';
+import { useDefault } from './default';
+import { useLoad } from './api';
+import { getNativeLocale } from '../api/nativelocale';
+
+export const useLocalizedPunctuation = () => {
+  // fallback to 'en-US' if native locale is an empty string
+  const locale = useDefault(useLoad(getNativeLocale), 'en-US') || 'en-US';
```

Comment:
this creates a bit too many requests to the native locale endpoint. Also currently the app doesn't listen or expect the locale to change during runtime.

I am thinking of adding this to appcontext provide @shonsirsha wdyt?

Reply:
actually I'll try to just use `i18n.langauge` that should also make the tests happy.